File: C++/src/item.cpp
#include "item.h"

Item::Item(int days_remaining, int quality)
    : _days_remaining(days_remaining)
    , _quality(quality)
{
}

void Item::tick()
{
}

int Item::days_remaining() const
{
    return _days_remaining;
}

int Item::quality() const
{
    return _quality;
}

File: Java/src/main/java/com/gildedrose/Item.java
package com.gildedrose;

public class Item {

    public String name;

    public int daysRemaining;

    public int quality;

    public Item(String name, int daysRemaining, int quality) {
        this.name = name;
        this.daysRemaining = daysRemaining;
        this.quality = quality;
    }

   @Override
   public String toString() {
        return this.name + ", " + this.daysRemaining + ", " + this.quality;
    }
}
